pkg/kube/controllers/extendedjob/job_reconciler_test.go

```diff
@@ -175,23 +179,39 @@ var _ = Describe("ReconcileExtendedJob", func() {
 			})
 		})
 
-		Context("when WriteOnFailure is not set", func() {
+		Context("when WriteOnFailure is set", func() {
 			JustBeforeEach(func() {
 				ejob.Spec.Output = &ejapi.Output{
 					NamePrefix:     "foo-",
 					WriteOnFailure: true,
 				}
+				pod2 = pod1.DeepCopy()
+				pod2.SetName("foo-job-latest")
+				pod2.SetCreationTimestamp(metav1.Now())
+				job.Status.Failed = 3
 			})
 
-			It("does persist the output", func() {
+			It("does persist the output of the latest pod", func() {
 				client.CreateCalls(func(context context.Context, object runtime.Object, _ ...crc.CreateOptionFunc) error {
 					secret := object.(*corev1.Secret)
 					Expect(secret.GetName()).To(Equal("foo-busybox"))
 					return nil
 				})
 
+				client.ListCalls(func(context context.Context, object runtime.Object, _ ...crc.ListOptionFunc) error {
+					switch object := object.(type) {
+					case *corev1.PodList:
+						list := corev1.PodList{
+							Items: []corev1.Pod{*pod1, *pod2},
+						}
+						list.DeepCopyInto(object)
+					}
+					return nil
+				})
+
 				result, err := reconciler.Reconcile(request)
 				Expect(err).ToNot(HaveOccurred())
+				Expect(logs.FilterMessageSnippet("Considering job pod foo-job-latest for persisting output").Len()).To(Equal(1))
```

Comment:
I think it would be good to not use a log message for verifying that the latest pod is being used for persistance but rather something lower level. Log messages tend to change and can lie ;)

Reply:
Changed it. Thanks for the solution.